src/python/pants/backend/go/util_rules/cgo_pkgconfig.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.go.util_rules import cgo_binaries
+from pants.backend.go.util_rules.cgo_binaries import CGoBinaryPathRequest
+from pants.backend.go.util_rules.cgo_security import (
+    check_compiler_flags,
+    check_linker_flags,
+    safe_arg,
+)
+from pants.core.util_rules.system_binaries import BinaryPath, BinaryPathTest
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import collect_rules, rule
+
+
+@dataclass(frozen=True)
+class CGoPkgConfigFlagsRequest:
+    """Request resolution of pkg-config arguments into CFLAGS and LDFLAGS."""
+
+    pkg_config_args: tuple[str, ...]
+
+
+@dataclass(frozen=True)
+class CGoPkgConfigFlagsResult:
+    cflags: tuple[str, ...]
+    ldflags: tuple[str, ...]
+
+
+# splitPkgConfigOutput parses the pkg-config output into a slice of
+# flags. This implements the algorithm from pkgconf/libpkgconf/argvsplit.c.
+def _split_pkg_config_output(content: bytes) -> tuple[str, ...]:
```

Comment:
Worth copying tests from Go itself? To make sure we don't unintentionally refactor this code and break things.

Reply:
Done.